tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Golden testing in Haskell with Hspec-Golden
+published: 2016-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golden-1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: A golden test is an IO action that writes its result to a file. To pass the test, this output file should be identical to the corresponding golden file, which contains the correct result for the test.
+---
+
+Golden tests are like unit tests, with the difference that the expected output is stored in a separate file. 
+
+Golden tests can also be known as:
+* Golden master testing
+* Approval testing
+* Snapshot testing
+* Expect testing
```

Comment:
@MedAdrian I'm not sure about that list here. We don't mention all of them in the rest of the blogpost, I think. We can probably merge it with the previous line:

> Golden testing (also be known as Golden master testing or Snapshot testing) is pretty similar to unit testing, except that the test output is stored in a separate file known as the "Golden File". 

What do you think?

Reply:
Sure thing! 👍